partner/partner-management-service/src/main/java/io/mosip/pms/oidc/client/service/impl/ClientManagementServiceImpl.java

```diff
@@ -0,0 +1,474 @@
+			throw new PartnerServiceException(ErrorCode.PARTNER_POLICY_NOT_APPROVED.getErrorCode(),
+					ErrorCode.PARTNER_POLICY_NOT_APPROVED.getErrorMessage());
+		}
+
+		ClientDetail clientDetail = new ClientDetail();
+		clientDetail.setPublicKey(getJWKString(createRequest.getPublicKey()));
+		clientDetail.setId(UUID.randomUUID().toString());
+		clientDetail.setName(createRequest.getName());
+		clientDetail.setRpId(createRequest.getRelayingPartyId());
+		clientDetail.setPolicyId(policyFromDb.getId());
+		clientDetail.setLogoUri(createRequest.getLogoUri());
+		clientDetail.setRedirectUris(String.join(",", createRequest.getRedirectUris()));
+		clientDetail.setClaims(String.join(",", authenticationContextClassRefUtil.getPolicySupportedClaims(getReqAttributeFromPolicyJson(
+				getPolicyObject(policyFromDb.getPolicyFileId()), ALLOWED_KYC_ATTRIBUTES, ATTRIBUTE_NAME, null))));
+		clientDetail.setAcrValues(String.join(",", authenticationContextClassRefUtil.getAuthFactors(getReqAttributeFromPolicyJson(
+				getPolicyObject(policyFromDb.getPolicyFileId()), ALLOWED_AUTH_TYPES, AUTH_TYPE, MANDATORY))));		
+		clientDetail.setStatus("ACTIVE");
+		clientDetail.setGrantTypes(String.join(",", createRequest.getGrantTypes()));
+		clientDetail.setClientAuthMethods(String.join(",", createRequest.getClientAuthMethods()));
+		clientDetail.setCreatedDateTime(LocalDateTime.now(ZoneId.of("UTC")));
+		clientDetail.setCreatedBy(getLoggedInUserId());
+		authenticationContextClassRefUtil.getPolicySupportedClaims(Arrays.asList(clientDetail.getClaims()));
+		callIdpService(clientDetail, environment.getProperty("pmp-idp.oidc.client.create.rest.uri"), true);
+		publishClientData(policyMappingReqFromDb.get(0).getPartner(), policyFromDb, clientDetail);
+		clientDetailRepository.save(clientDetail);
+		var response = new ClientDetailResponse();
+		response.setClientId(clientDetail.getId());
+		response.setStatus(clientDetail.getStatus());		
+		return response;
+	}
+
+	/**
+	 * 
+	 * @param policyObject
+	 * @param parentAttribute
+	 * @param childAttribute
+	 * @param childAttribute1
+	 * @return
+	 * @throws Exception
+	 */
+	private static List<String> getReqAttributeFromPolicyJson(JSONObject policyObject, String parentAttribute,
+			String childAttribute, String childAttribute1) throws Exception {
+		List<String> attributes = new ArrayList<>();
+		JSONArray parentAttributeObject = (JSONArray) policyObject.get(parentAttribute);
+		for (int i = 0; i < parentAttributeObject.size(); i++) {
+			JSONObject childJsonArray = (JSONObject) parentAttributeObject.get(i);
+			String key = (String) childJsonArray.get(childAttribute);
+			if (childAttribute1 != null) {
+				if ((boolean) childJsonArray.get(childAttribute1)) {
+					LOGGER.error("createOIDCClient::Policy is having mandatory auth. So client cannot be created.");
+					throw new PartnerServiceException(ErrorCode.POLICY_HAVING_MANDATORY_AUTHS.getErrorCode(),
```

Comment:
Little confused what are we trying to do here? Why is there no comment on what this function is about